### PR TITLE
Cloudbuild logs bucket

### PR DESCRIPTION
**Pull request by paulmattei**
Send Cloudbuild logs to GCS storage bucket 'trillian-cloudbuild-logs'

### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
`Your build failed to run: generic::invalid_argument: invalid bucket "trillian-cloudbuild-logs"; builder service account does not have access to the bucket`

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@
  * `testdb` no longer accepts the `--test_mysql_uri` flag, and instead honours the
    `TEST_MYSQL_URI` ENV var. This makes it easier to blanket configure tests to use a
    specific test DB instance.
+ * Set GCS bucket to send build logs to in cloudbuild.yaml
```
Comment (pavelkalinnikov) [inline CHANGELOG.md:23]: This isn't necessary to be in the CHANGELOG, it will covered by "migrated to Cloud Build" item.

File: cloudbuild.yaml [hunk 2]
```diff
@@ -23,6 +23,9 @@ options:
 # TODO(pavelkalinnikov): Consider pushing this image only on commits to master.
 images: ['gcr.io/$PROJECT_ID/trillian_testbase:latest']
 
+# Cloud Build logs sent to GCS bucket
+logsBucket: 'gs://trillian-cloudbuild-logs'
```
Comment (pavelkalinnikov) [inline cloudbuild.yaml:27]: The bucket name in GCS has underscores instead of dashes.

Comment (paulmattei) [reply cloudbuild.yaml:27]: Created 'trillian-cloudbuild-logs'

File: CHANGELOG.md (hunk 1 above)
Comment (paulmattei) [reply CHANGELOG.md]: Removed

**Review by pavelkalinnikov — APPROVED** [review]

**Pull request merged**